sdk/python/feast/infra/online_stores/snowflake.py

```diff
@@ -0,0 +1,232 @@
+import itertools
+import os
+from binascii import hexlify
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple
+
+import pandas as pd
+import pytz
+from pydantic import Field
+from pydantic.schema import Literal
+
+from feast import Entity, FeatureView
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.infra.utils.snowflake_utils import get_snowflake_conn, write_pandas_binary
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import FeastConfigBaseModel, RepoConfig
+from feast.usage import log_exceptions_and_usage
+
+
+class SnowflakeOnlineStoreConfig(FeastConfigBaseModel):
+    """ Online store config for Snowflake """
+
+    type: Literal["snowflake.online"] = "snowflake.online"
+    """ Online store type selector"""
+
+    config_path: Optional[str] = (
+        Path(os.environ["HOME"]) / ".snowsql/config"
+    ).__str__()
+    """ Snowflake config path -- absolute path required (Can't use ~)"""
+
+    account: Optional[str] = None
+    """ Snowflake deployment identifier -- drop .snowflakecomputing.com"""
+
+    user: Optional[str] = None
+    """ Snowflake user name """
+
+    password: Optional[str] = None
+    """ Snowflake password """
+
+    role: Optional[str] = None
+    """ Snowflake role name"""
+
+    warehouse: Optional[str] = None
+    """ Snowflake warehouse name """
+
+    database: Optional[str] = None
+    """ Snowflake database name """
+
+    schema_: Optional[str] = Field("PUBLIC", alias="schema")
+    """ Snowflake schema name """
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class SnowflakeOnlineStore(OnlineStore):
```

Comment:
Can you add comments on the rough data model being used in this online store?

Seems like it's mainly there is a new table for each feature view, there is no primary key, and there's some entity_feature_key?

Reply:
Same data model is SQLite -- snowflake regular tables dont need a primary key.